pkg/skaffold/deploy/util/namespaces.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2021 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"sort"
+
+	kubectx "github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubernetes/context"
+	latestV1 "github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest/v1"
+)
+
+// GetAllPodNamespaces lists the namespaces that should be watched.
+// + The namespace passed on the command line
+// + Current kube context's namespace
+// + Namespaces referenced in Helm releases
+func GetAllPodNamespaces(configNamespace string, pipelines []latestV1.Pipeline) ([]string, error) {
+	nsMap := make(map[string]bool)
+
+	if configNamespace == "" {
+		// Get current kube context's namespace
+		config, err := kubectx.CurrentConfig()
+		if err != nil {
+			return nil, fmt.Errorf("getting k8s configuration: %w", err)
+		}
+
+		context, ok := config.Contexts[config.CurrentContext]
+		if ok {
+			nsMap[context.Namespace] = true
+		} else {
+			nsMap[""] = true
+		}
+	} else {
+		nsMap[configNamespace] = true
+	}
+
+	// Set additional namespaces each helm release referenced
+	for _, namespace := range collectHelmReleasesNamespaces(pipelines) {
```

Comment:
This is a bit grotty: we shouldn't embed Helm knowledge here.  Why not move this collectHelm* function into pkg/skaffold/deploy/helm and instead pass in the additional namespaces as an argument.

(And I thought if we used `skaffold deploy -n xxx` then `xxx` [overrides all release namespaces](https://github.com/GoogleContainerTools/skaffold/blob/master/pkg/skaffold/deploy/helm/args.go#L156-L158)?)

Reply:
sorry I should have specified - this code is being added back from [the offending PR](https://github.com/GoogleContainerTools/skaffold/pull/6170/files#diff-440d39b3f21cbf92501acb97d6ada497928b15e0084537f5428fbc69b2c27f46L31) which removed it. I did not write this - it just seems safer to add it back as it was, and then make modifications to it later.

Reply:
re: your comment about Helm - I think you're right, our namespace-handling code was (and still is) a bit of a mess. I'm open to cleaning it up, I just don't think we should do it here.